### PR TITLE
Qradar bug fix

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/20776

## Description
Fixed an issue in get-assets command that when you enter 'fields' param the command failed

## Screenshots
image

## Does it break backward compatibility?
   - No

### CODE REVIEW

File: TestPlaybooks/playbook-TestQradar.yml [hunk 1]
```diff
@@ -1,15 +1,14 @@
 id: test_Qradar
 version: -1
 name: test_Qradar
-fromversion: 4.1.0
```
Comment (Itay4) [inline TestPlaybooks/playbook-TestQradar.yml]: revert this

File: Integrations/QRadar/QRadar.py [hunk 2]
```diff
@@ -792,31 +795,34 @@ def transform_single_asset_to_hr(asset):
...
-            asset_dict[ASSET_PROPERTIES_NAMES_MAP[prop['name']]] = {'Value': prop['value'],
-                                                                    'LastUser': prop['last_reported_by']}
-        elif prop['name'] in ASSET_PROPERTIES_ENDPOINT_NAMES_MAP:
-            endpoint_dict[ASSET_PROPERTIES_ENDPOINT_NAMES_MAP[prop['name']]] = prop['value']
-        elif full_values:
-            if prop['name'] in FULL_ASSET_PROPERTIES_NAMES_MAP:
-                asset_dict[FULL_ASSET_PROPERTIES_NAMES_MAP[prop['name']]] = {'Value': prop['value'],
-                                                                             'LastUser': prop['last_reported_by']}
+    if 'properties' in asset:
+        for prop in asset['properties']:
```
Comment (Itay4) [inline Integrations/QRadar/QRadar.py]:
```suggestion
        for prop in asset.get('properties', []):
```

File: Integrations/QRadar/QRadar.py [hunk 3]
```diff
@@ -792,31 +795,34 @@ def transform_single_asset_to_hr(asset):
...
 def create_single_asset_result_and_enrich_endpoint_dict(asset, endpoint_dict, full_values):
-    asset_dict = {'ID': asset['id']}
-    for interface in asset['interfaces']:
-        if full_values:
-            endpoint_dict['MACAddress'].append(interface['mac_address'])
-        for ip_address in interface['ip_addresses']:
-            endpoint_dict['IPAddress'].append(ip_address['value'])
+    asset_dict = {'ID': asset.get('id')}
+    if 'interfaces' in asset:
+        for interface in asset['interfaces']:
```
Comment (Itay4) [inline Integrations/QRadar/QRadar.py]:
```suggestion
        for interface in asset.get('interfaces', []):
```

File: Integrations/QRadar/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 ## [Unreleased]
-
+-
```
Comment (Itay4) [inline Integrations/QRadar/CHANGELOG.md]: why is the changelog empty?

File: Integrations/QRadar/QRadar.py [hunk 5]
```diff
@@ -792,31 +795,34 @@ def transform_single_asset_to_hr(asset):
...
-                                                                    'LastUser': prop['last_reported_by']}
-        elif prop['name'] in ASSET_PROPERTIES_ENDPOINT_NAMES_MAP:
-            endpoint_dict[ASSET_PROPERTIES_ENDPOINT_NAMES_MAP[prop['name']]] = prop['value']
-        elif full_values:
-            if prop['name'] in FULL_ASSET_PROPERTIES_NAMES_MAP:
-                asset_dict[FULL_ASSET_PROPERTIES_NAMES_MAP[prop['name']]] = {'Value': prop['value'],
-                                                                             'LastUser': prop['last_reported_by']}
+    if 'properties' in asset:
+        for prop in asset['properties']:
+            if prop['name'] in ASSET_PROPERTIES_NAMES_MAP:
```
Comment (Itay4) [inline Integrations/QRadar/QRadar.py]: there are also direct key access in these conditions as well, use `.get()` instead

File: Integrations/QRadar/QRadar.py [hunk 6]
```diff
@@ -797,32 +797,33 @@ def transform_single_asset_to_hr(asset):
 def create_single_asset_result_and_enrich_endpoint_dict(asset, endpoint_dict, full_values):
     asset_dict = {'ID': asset.get('id')}
     if 'interfaces' in asset:
```
Comment (Itay4) [inline Integrations/QRadar/QRadar.py]: remove this, no longer needed

File: Integrations/QRadar/QRadar.py [hunk 7]
```diff
@@ -797,32 +797,33 @@ def transform_single_asset_to_hr(asset):
...
-            endpoint_dict['Domain'].append(domain_name)
+            domain_name = get_domain_name(asset.get('domain_id'))
+            endpoint_dict.get('Domain').append(domain_name)
     # Adding values found in properties of the asset
     enrich_dict_using_asset_properties(asset, asset_dict, endpoint_dict, full_values)
     return asset_dict
 
 
 def enrich_dict_using_asset_properties(asset, asset_dict, endpoint_dict, full_values):
     if 'properties' in asset:
```
Comment (Itay4) [inline Integrations/QRadar/QRadar.py]: remove this, no longer needed

**Comment by Shellyber** [comment]
@Itay4 For the `QRadarFullSearch` playbook, it runs also in the test and passed with no issues

**Comment by kirbles19** [comment]
@Shellyber Done.

**Pull request merged**